tests/block_translator.rs

```diff
@@ -0,0 +1,152 @@
+use core::array;
+use embedded_io::ErrorType;
+use simple_fatfs::block_io::{BlockBase, BlockIndex, BlockRead, BlockTranslator, BlockWrite};
+
+#[derive(Debug)]
+struct Storage<'a, const BS: usize>(&'a mut [u8; 64]);
+
+impl<const BS: usize> BlockBase for Storage<'_, BS> {
+    fn block_size(&self) -> usize {
+        BS
+    }
+
+    fn block_count(&self) -> usize {
+        64 / BS
+    }
+}
+
+impl<const BS: usize> BlockRead for Storage<'_, BS> {
+    fn read(&mut self, block: BlockIndex, buf: &mut [u8]) -> Result<(), Self::Error> {
+        let offset = block as usize * BS;
+        buf.copy_from_slice(&self.0[offset..offset + buf.len()]);
+        Ok(())
+    }
+}
+
+impl<const BS: usize> ErrorType for Storage<'_, BS> {
+    type Error = embedded_io::SliceWriteError;
+}
+
+impl<const BS: usize> BlockWrite for Storage<'_, BS> {
+    fn write(&mut self, block: BlockIndex, buf: &[u8]) -> Result<(), Self::Error> {
+        let offset = block as usize * BS;
+        self.0[offset..offset + buf.len()].clone_from_slice(&buf);
+        Ok(())
+    }
+
+    fn flush(&mut self) -> Result<(), Self::Error> {
+        Ok(())
+    }
+}
+
+#[test]
+fn test_block_translator1() {
+    let mut translated_c_buffer1 = [0u8; 4];
+
+    run_block_translator([&mut translated_c_buffer1]);
+}
+
+#[test]
+fn test_block_translator2() {
+    let mut translated_c_buffer1 = [0u8; 4];
+    let mut translated_c_buffer2 = [0u8; 4];
+
+    run_block_translator([&mut translated_c_buffer1, &mut translated_c_buffer2]);
+}
+
+#[test]
+fn test_block_translator3() {
+    let mut translated_c_buffer1 = [0u8; 4];
+    let mut translated_c_buffer2 = [0u8; 4];
+    let mut translated_c_buffer3 = [0u8; 4];
+
+    run_block_translator([
+        &mut translated_c_buffer1,
+        &mut translated_c_buffer2,
+        &mut translated_c_buffer3,
+    ]);
+}
+
+#[test]
+fn test_block_translator8() {
+    let mut translated_c_buffer1 = [0u8; 4];
+    let mut translated_c_buffer2 = [0u8; 4];
+    let mut translated_c_buffer3 = [0u8; 4];
+    let mut translated_c_buffer4 = [0u8; 4];
+    let mut translated_c_buffer5 = [0u8; 4];
+    let mut translated_c_buffer6 = [0u8; 4];
+    let mut translated_c_buffer7 = [0u8; 4];
+    let mut translated_c_buffer8 = [0u8; 4];
+
+    run_block_translator([
+        &mut translated_c_buffer1,
+        &mut translated_c_buffer2,
+        &mut translated_c_buffer3,
+        &mut translated_c_buffer4,
+        &mut translated_c_buffer5,
+        &mut translated_c_buffer6,
+        &mut translated_c_buffer7,
+        &mut translated_c_buffer8,
+    ]);
+}
```

Comment:
Just for aesthetics, this could be replaced with `core::array::repeat`. This function however was only added to Rust last month, so that means we would have to add a `rust-version = "1.91.0"` field to `Cargo.toml`. In my opinion it isn't worth it for such a small change to make that change, but if you want to, go ahead.

Reply:
I'd keep it like this until the msrv is 1.91 because otherwise there will be this and another case...